### PR TITLE
Solved the task

### PR DESCRIPTION
**Pull request by Cerne13**

### CODE REVIEW

File: services/user.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    if email or first_name or last_name:
+        user.save()
+
+    return user
+
+
+def get_user(user_id: int) -> User:
+    user = User.objects.get(id=user_id)
+    return user
```
Comment (YevheniiDavydenko) [inline services/user.py:33]: you can solve it in one line without user

File: services/order.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                order_id=order.id,
+                row=ticket["row"],
+                seat=ticket["seat"]
+            )
+
+        return order
+
+
+def get_orders(username: str = None) -> QuerySet[Order]:
+    orders = Order.objects.all().prefetch_related("user")
```
Comment (YevheniiDavydenko) [inline services/order.py:32]: read checklist in readme, you don't need all(). here

File: services/order.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            order.created_at = date
+            order.save()
+
+        for ticket in tickets:
+            Ticket.objects.create(
+                movie_session_id=ticket["movie_session"],
+                order_id=order.id,
+                row=ticket["row"],
+                seat=ticket["seat"]
+            )
```
Comment (YevheniiDavydenko) [inline services/order.py:26]: I think that bulk_create will run much faster
also you can use list comprehesions

File: db/models.py [hunk 4]
```diff
@@ -45,3 +55,79 @@ class MovieSession(models.Model):
...
+        self.full_clean()
+
+        return super(Ticket, self).save(
+            force_insert,
+            force_update,
+            using,
+            update_fields
+        )
+
+    class Meta:
```
Comment (YevheniiDavydenko) [inline db/models.py:127]: I don't now, but by my opinion it will be better for readability if you put Meta right after attributes

**Review by YevheniiDavydenko — CHANGES_REQUESTED** [review]

File: services/order.py (hunk 2 above)
Comment (Cerne13) [reply services/order.py:32]: Thank you. Was constantly worried about electricity, so made a couple of stupid mistakes :)

File: db/models.py (hunk 4 above)
Comment (Cerne13) [reply db/models.py:127]: Yes, I agree. Somehow thought that typical order is 'attrs -> methods -> meta'.
Just checked. As far as seen, the best practice is considered 'attrs -> meta -> methods'

File: services/order.py (hunk 3 above)
Comment (Cerne13) [reply services/order.py:26]: Oh, yes. Missed this moment :)
Remembered later, but hadn't time to fix.

**Review by zhulich — APPROVED** [review]
GJ!

File: services/movie_session.py [hunk 5]
```diff
@@ -42,3 +42,15 @@ def update_movie_session(
...
+
+def get_taken_seats(movie_session_id: int) -> list[dict]:
+    session = MovieSession.objects.get(id=movie_session_id)
+
+    return [
+        {
+            "row": ticket.row,
+            "seat": ticket.seat
+        }
+        for ticket in session.ticket_set.all()
```
Comment (TPolina) [inline services/movie_session.py:55]: better to add and use related_name

File: services/order.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            [Ticket(
+                movie_session=MovieSession.objects.get(
+                    id=ticket["movie_session"]
+                ),
+                order_id=order.id,
+                row=ticket["row"],
+                seat=ticket["seat"]
+            ) for ticket in tickets
+            ]
+        )
```
Comment (TPolina) [inline services/order.py:30]: why not iterate trough tickets and create each separately? Your code would be much more neat

File: services/order.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        order = Order.objects.create(user=user)
+
+        if date:
+            order.created_at = date
+            order.save()
+
+        Ticket.objects.bulk_create(
+            [Ticket(
+                movie_session=MovieSession.objects.get(
+                    id=ticket["movie_session"]
```
Comment (TPolina) [inline services/order.py:23]: you don't need to get `MovieSession` object you can use `movie_session_id`

File: services/user.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    )
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    if email or first_name or last_name:
```
Comment (TPolina) [inline services/user.py:25]: you can omit this check

**Review by TPolina — CHANGES_REQUESTED** [review]

File: services/movie_session.py (hunk 5 above)
Comment (Cerne13) [reply services/movie_session.py:55]: Thank you

File: services/order.py (hunk 7 above)
Comment (Cerne13) [reply services/order.py:23]: There were some problems with id during bulk creation. I'll definitely change it back to session id when I change the method to create()

**Review by TPolina — APPROVED** [review]
Well done!

**Pull request closed without merging**